README.md

```diff
@@ -128,6 +123,16 @@ pre-commit run --all-files
 
 ### Testing
 
+There are two types of tests: mocking the PyFirecREST or the FirecREST server.
+While the former is a good practice to ensure that all three (`aiida-firecrest`, FirecREST, and PyFirecREST) work flawlessly, debugging may not always be easy because it may not always be obvious which of the three is causing a bug.
```

Comment:
```suggestion
While the latter is a good practice to ensure that all three (`aiida-firecrest`, FirecREST, and PyFirecREST) work flawlessly, debugging may not always be easy because it may not always be obvious which of the three is causing a bug.
```

I think this sentence refers to mocking the FireCREST server, right. I also think it is a matter of opinion whether you consider this "good practice". Usually testing resources tell you not to test internals of dependencies.